feat(APP-279): Add Sentry environments and sanitizer for logging of environment variables during builds

tyhonchik opened the pull request:
# Description

Added proper Sentry envs to building CI stage.
Added logging and sanitizing of envs for debug and observability purposes

## Type of Change

<!--- Please delete the options that are not relevant. -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Developer Checklist:

- [x] Manually smoke tested the functionality locally
- [x] Confirmed there are no new warnings or errors in the browser console
- [ ] (For User Stories only) Double-checked that all Acceptance Criteria are satisfied
- [x] Confirmed there are no new warnings on automated tests
- [x] Merged and published any dependent changes in downstream modules
- [x] Selected the correct base branch
- [x] Commented the code in hard-to-understand areas
- [x] Followed the code style guidelines of this project
- [x] Reviewed that the Files Changed in Github’s UI reflect my intended changes
- [ ] Confirmed the pipeline checks are not failing

## Review Checklist:

- [ ] Tested locally that all Acceptance Criteria or Expected Outcomes are satisfied
- [ ] Confirmed that changes follow the code style guidelines of this project

Fabricevladimir commented on packages/web-app/src/context/createProposal.tsx:
```diff
@@ -483,7 +483,10 @@ const CreateProposalProvider: React.FC<Props> = ({
         daoName: daoDetails?.metadata.name,
         proposalGuid,
-        proposalParams: proposalCreationData,
+        proposalParams: {
```
Can you port this over to the Settings proposal creation wrapper as well?

Fabricevladimir reviewed (commented):
Good stuff!

tyhonchik committed:
feat(APP-279): Add environment variable sanitizer and update Sentry configuration

- Introduced an environment variable sanitizer to log sanitized environment variables during builds.
- Updated Sentry configuration.
- Adjusted sourcemap upload behavior).
- Enhanced GitHub Actions workflows to inherit secrets.

milosh86 commented on .github/workflows/shared-build.yml:
```diff
@@ -37,6 +37,11 @@ jobs:
+        env:
+          NEXT_PUBLIC_SENTRY_ORG: ${{ vars.NEXT_PUBLIC_SENTRY_ORG }}
+          NEXT_PUBLIC_SENTRY_PROJECT: ${{ vars.NEXT_PUBLIC_SENTRY_PROJECT }}
+          NEXT_PUBLIC_SENTRY_DSN: ${{ secrets.NEXT_PUBLIC_SENTRY_DSN }}
```
It's probably nothing but here we have `NEXT_PUBLIC` var in `secrets` ... is that intended?

tyhonchik replied:
yep, basically it is public
but I wanted to hide it a bit

milosh86 commented on next.config.mjs:
```diff
@@ -1,6 +1,9 @@
 import packageInfo from './package.json' with { type: 'json' };
+import envSanitizer from './scripts/dumpEnvSanitized.mjs';
+
+envSanitizer.dump();
```
hm, not sure about this, even with protection it provides .... Is it really required? 
I would rather try to log cherry picked vars that might be useful seeing in logs and avoid just dumping everything.

tyhonchik replied:
I added this for debugging and decided to leave it -- it's useful to see what variables the application built and launched with.

But you're right -- this isn't where this log should be. I'll create a separate task to log environment variables in Sentry or Versel or somewhere else

tyhonchik replied:
created
relates to [APP-313](https://linear.app/aragon/issue/APP-313/log-runtime-environment-variables-to-sentry-for-better-observability)

milosh86 reviewed (commented):
Hey, looks good overall, I just left a comment about the concern I have about env vars dump. Let's try to avoid it and let's instead start a discussion on how to simplify env vars management so it's more accessible.

tyhonchik committed:
fix: Remove env logging and sanitizing

milosh86 reviewed (approved):
looks good 👍
